packages/ui/src/forum/modals/CreateThreadModal/machine.ts

```diff
@@ -0,0 +1,61 @@
+import { createMachine, assign } from 'xstate'
+
+import { EmptyObject } from '@/common/types'
+
+export interface GeneralFormContext {
+  topic: string
+  description: string
+}
+
+type TransactionContext = Required<GeneralFormContext>
+
+type CreateThreadContext = Partial<TransactionContext>
+
+type CreateThreadState =
+  | { value: 'requirementsVerification'; context: EmptyObject }
+  | { value: 'requirementsFailed'; context: EmptyObject }
+  | { value: 'generalDetails'; context: GeneralFormContext }
+  | { value: 'end'; context: TransactionContext }
+
+export type CreateThreadEvent =
+  | { type: 'FAIL' }
+  | { type: 'NEXT' }
+  | { type: 'BACK' }
+  | { type: 'SET_TOPIC'; topic: string }
+  | { type: 'SET_DESCRIPTION'; description: string }
+
+export const createThreadMachine = createMachine<CreateThreadContext, CreateThreadEvent, CreateThreadState>({
+  initial: 'requirementsVerification',
+  context: {
+    topic: '',
+    description: '',
+  },
+  states: {
+    requirementsVerification: {
+      on: {
+        NEXT: 'generalDetails',
+        FAIL: 'requirementsFailed',
+      },
+    },
+    requirementsFailed: { type: 'final' },
+    generalDetails: {
+      on: {
+        NEXT: {
+          target: 'end',
+          cond: (context) => !!(context['topic'] && context['description']),
```

Comment:
```suggestion
          cond: (context) => !!(context.topic && context.description),
```

Usually we use the normal accessor for that and we use array access when it is not a valid JS name, like `context['some-foo']`.